ext/zkrb.c

```diff
@@ -73,6 +73,17 @@
 #include "ruby/io.h"
 #endif
 
+#ifndef HAVE_RB_THREAD_FD_SELECT
+#define rb_fdset_t fd_set
+#define rb_fd_isset(n, f) FD_ISSET(n, f)
+#define rb_fd_init(f)
+#define rb_fd_zero(f)  FD_ZERO(f)
```

Comment:
hrm, [here](https://github.com/ruby/ruby/blob/ruby_2_2/include/ruby/intern.h#L362) it seems to be. 
don't quite understand what's going on [here](https://github.com/ruby/ruby/blob/ruby_2_2/thread.c#L3348)

It would definitely seem that rb_fd_init() being a no-op instead of calling FD_ZERO is a definite bug, and would cause the behavior I saw.